pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java

```diff
@@ -22,6 +25,28 @@
 
   private static final Logger LOGGER = Logger.getLogger(BooleanTypeUtil.class.getName());
 
+  /**
+   * Maps String values to Boolean constants. By using {@link String#CASE_INSENSITIVE_ORDER}
+   * {@link Map#get(Object)} is case-insensitive.
+   */
+  private static final SortedMap<String, Boolean> STRING_TO_BOOL = new TreeMap<String, Boolean>(
```

Comment:
I have not compared, but currently to get to "false" requires at least 2 equals and 6 equal ignore case comparisons. It could also take up to 5 additional equal ignore case comparisons.
By using a TreeMap, it now becomes effectively a binary search ignoring case.